Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -103,6 +107,18 @@ cancel bookingId _ req = do
           Right res' -> do
             let merchantOperatingCityId = booking.merchantOperatingCityId
             disToPickup <- driverDistanceToPickup booking.merchantId merchantOperatingCityId (getCoordinates res'.currPoint) (getCoordinates booking.fromLocation)
+            now <- getCurrentTime
+            let driverBookingDuration = roundToIntegral $ diffUTCTime now booking.createdAt
+            distanceRes <-
+              Maps.getDistance booking.merchantId merchantOperatingCityId $
+                Maps.GetDistanceReq
+                  { origin = ride.driverInitialLocation,
+                    destination = getCoordinates res'.currPoint,
+                    travelMode = Just Maps.CAR
+                  }
+            person <- QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+            when ((driverBookingDuration > merchant.driverBookingTime && disToPickup < round merchant.driverDistanceToPickup) || (realToFrac distanceRes.distance) > merchant.driverPickupDistance) $ do
```

Comment:
@JanhaviSoni What is this logic?

Reply:
These are the conditions for cancellation

Plz find this doc @hemantmangla 
https://docs.google.com/document/d/1D5vU3gQ_NFUqFncNbbSqb2ahf9zjcnZzytdt6tXnrZ4/edit